packages/api/src/module/write/user-registration/presentation/rest/user-registration.rest-controller.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { HttpStatus } from '@nestjs/common';
+import { AsyncReturnType } from 'type-fest';
+
+import { registerError } from '@coderscamp/shared/models/auth/register';
+
+import { DomainRuleViolationException } from '@/shared/errors/domain-rule-violation.exception';
+import { initTestModuleRestApi } from '@/shared/test-utils';
+import { UserRegistrationRestController } from '@/write/user-registration/presentation/rest/user-registration.rest-controller';
+
+import { initOpenApiExpect } from '../../../../../../jest-setup';
+
+initOpenApiExpect();
+
+// fixme: ADDRESS ALREADY IN USE
+// eslint-disable-next-line jest/no-disabled-tests
+describe.skip('User Registration | REST API', () => {
+  let restUnderTest: AsyncReturnType<typeof initTestModuleRestApi>;
+
+  beforeAll(async () => {
+    restUnderTest = await initTestModuleRestApi(UserRegistrationRestController);
+    restUnderTest.commandBusExecute.mockClear();
+  });
+
+  afterAll(async () => {
+    await restUnderTest.close();
+  });
+
+  describe('/POST user-registration', () => {
+    it(`success`, async () => {
+      // Given
+      restUnderTest.commandBusExecute.mockImplementation(() => Promise.resolve());
+
+      // When
+      const response = await restUnderTest.http.post('/api/user-registration').send({
+        fullName: 'John Kowalsky',
+        email: 'jan.kowalski@nowak.pl',
+        password: 'samplePassword123',
+      });
+
+      // Then
+      expect(response.status).toBe(HttpStatus.CREATED);
+      expect(response.body.userId).toBeDefined();
+      expect(response).toSatisfyApiSpec();
+    });
+
+    it(`invalid request body - without password field`, async () => {
+      // When
+      const response = await restUnderTest.http.post('/api/user-registration').send({
+        fullName: 'John Kowalsky',
+        email: 'jan.kowalski@nowak.pl',
+      });
+
+      // Then
+      expect(response.status).toBe(HttpStatus.BAD_REQUEST);
+      expect(response.body).toStrictEqual({
+        statusCode: HttpStatus.BAD_REQUEST,
+        message: ['"password" is required', '"password" must be a string'],
+        error: 'Bad Request',
+      });
+      expect(response).toSatisfyApiSpec();
+    });
+
+    it(`command rejected because REGISTRATION_FORM_ALREADY_EXISTS`, async () => {
+      // Given
+      restUnderTest.commandBusExecute.mockRejectedValue(
+        new DomainRuleViolationException(registerError.USER_WAS_ALREADY_REGISTERED),
+      );
+
+      // When
+      const response = await restUnderTest.http.post('/api/user-registration').send({
+        fullName: 'John Kowalsky',
+        email: 'jan.kowalski@nowak.pl',
+        password: 'samplePassword123',
+      });
+
+      // Then
+      expect(response.status).toBe(HttpStatus.BAD_REQUEST);
+      expect(response.body).toStrictEqual({
+        statusCode: HttpStatus.BAD_REQUEST,
+        message: 'USER_WAS_ALREADY_REGISTERED',
+        error: 'Bad Request',
+      });
+      expect(response).toSatisfyApiSpec();
```

Comment:
Does anyone know what to do for TS to recognize custom matchers? 
![image](https://user-images.githubusercontent.com/18637116/133394640-ab4353e1-047f-4b9d-9b71-b893e0d26c5b.png)